packages/linuxheaders.rb

```diff
@@ -3,49 +3,51 @@
 class Linuxheaders < Package
   description 'Linux headers for Chrome OS.'
   homepage 'https://kernel.org/'
-  # KERNEL_VERSION = %x[uname -r].chomp.reverse.split('.',2).collect(&:reverse)[1]
-  case ARCH
-  when 'aarch64', 'armv7l', 'x86_64'
-    @_ver = '4.14'
-    version "#{@_ver}-1"
-  when 'i686'
-    @_ver = '3.8'
-    version @_ver
-  end
-  # Only check for kernel version if not in container.
-  unless File.exist?('/.dockerenv')
-    @KERNEL_VERSION = `uname -r`.chomp.reverse.split('.', 2).collect(&:reverse)[1]
-    @ver = @KERNEL_VERSION.between?(@KERNEL_VERSION, '5.15') ? @ver : @KERNEL_VERSION
-    version @ver
-  end
+  @version = CREW_KERNEL_VERSION == '4.14' ? "#{CREW_KERNEL_VERSION}-1" : CREW_KERNEL_VERSION
+  set_property('version', @version)
```

Comment:
The `version` function doesn't work for this. I tried it!

I'd be so happy if I were wrong. Feel free to show me otherwise!

This code does not work: `version = CREW_KERNEL_VERSION == '4.14' ? "#{CREW_KERNEL_VERSION}-1" : CREW_KERNEL_VERSION`

Reply:
Do a `CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=armv7l CREW_TESTING=1 crew update`, then edit linuxheaders.rb to just have `version = CREW_KERNEL_VERSION == '4.14' ? "#{CREW_KERNEL_VERSION}-1" : CREW_KERNEL_VERSION`, and then try `crew update compatible -v | grep linuxheaders`...

Reply:
`version` is a function, not a variable...

This should work:
```ruby
version CREW_KERNEL_VERSION == '4.14' ? "#{CREW_KERNEL_VERSION}-1" : CREW_KERNEL_VERSION
```

Reply:
As mentioned above, that does NOT work! (Maybe something broke at some point?)
With that change, I see this. (`crew update compatible -v` shows the version number.)
```
crew update compatible -v | grep linuxheaders
Checking linuxheaders for compatibility.
Adding linuxheaders  to compatible packages.
Checking musl_linuxheaders for compatibility.
Adding musl_linuxheaders 4.14 to compatible packages.
```
In contrast, this is what I see using `set_property('version', @version)`:
```
chronos@rpi4b-armv7l /usr/local/lib/crew/packages $ crew update compatible -v | grep linuxheaders
Checking linuxheaders for compatibility.
Adding linuxheaders 4.14-1 to compatible packages.
Checking musl_linuxheaders for compatibility.
Adding musl_linuxheaders 4.14 to compatible packages.
```

Reply:
I think the issue might be the `version` property acts like a constant and once set, cannot be changed.

Reply:
> I think the issue might be the `version` property acts like a constant and once set, cannot be changed.

But then `version CREW_KERNEL_VERSION == '4.14' ? "#{CREW_KERNEL_VERSION}-1" : CREW_KERNEL_VERSION` should work, since it is setting the constant once.

From my investigations using that code, and similar code using a case block, the constant never actually gets set.
I had to adapt the code from https://stackoverflow.com/questions/7849521/set-attribute-dynamically-of-ruby-object/39063481#39063481 to get this to work... which involved treating `version` as an attribute which needs to have the property set dynamically.